dotnet/src/SemanticKernel.Core/Data/TextSearch/TextSearchServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.Data;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Extension methods to register <see cref="ITextSearch"/> for use with <see cref="IServiceCollection"/>.
+/// </summary>
+public static class TextSearchServiceCollectionExtensions
+{
+    /// <summary>
+    /// Register an <see cref="ITextSearch"/> instance with the specified service ID.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="ITextSearch"/> on.</param>
+    /// <param name="textSearch">Instance of <see cref="ITextSearch"/> to register.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    public static IServiceCollection AddTextSearch(
+        this IServiceCollection services,
+        ITextSearch textSearch,
+        string? serviceId = default)
+    {
+        ITextSearch Factory(IServiceProvider serviceProvider, object? _) => textSearch;
+        services.AddKeyedSingleton<ITextSearch>(serviceId, (Func<IServiceProvider, object?, ITextSearch>)Factory);
```

Comment:
Consider this overload, which allows you to just pass the required instance without needing a factory when registering a singleton:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addkeyedsingleton?view=net-8.0#microsoft-extensions-dependencyinjection-servicecollectionserviceextensions-addkeyedsingleton-1(microsoft-extensions-dependencyinjection-iservicecollection-system-object-0)